deploy/crds/litmuschaos_v1alpha1_chaosexperiment_cr.yaml

```diff
@@ -1,7 +1,43 @@
-apiVersion: litmuschaos.io/v1alpha1
+---
+## An experiment is the definition of a chaos test and is listed as an item 
+## in the chaos engine to be run against a given app. Consists of reference
+## to chaosGraph (which has chaos param details and executor spec) and also
+## component information for the Object-Under-Test. The results upon running
+## this experiment is relayed back to the chaos engine which invoked it.
+
+## Experiments can be pulled from litmus charts and are versioned based on 
+## chart.
+
+## Charts should (not?) be auto-upgraded
+
+apiVersion: litmus.io/v1alpha1
 kind: ChaosExperiment
 metadata:
-  name: example-chaosexperiment
+
+  ## Eventually launched chaos litmusbook/job will bear <name>-<hash>
+  name: disappearing-pods
+  namespace: nginx
+
 spec:
-  # Add fields here
-  size: 3
+
+  chart: 
+    - name: kubernetes
```

Comment:
Yes, above fits my thought process as well. Experiments, if belonging/pulled by a chart -> the chaosGraph/Type also gets pulled as part of it in an implict way - because of the 1-1 mapping between them.

Reply:
It should be OK for the experiment to reuse an "older" or "newer" template, i.e., w/ a different chart version - as a backward/forward compatibility feature. This is feasible as the executor is embedded into the chaosTemplate itself!